packages/flutter_tools/gradle/kt/src/test/kotlin/com/flutter/gradle/BaseApplicationNameHandlerTest.kt

```diff
@@ -0,0 +1,60 @@
+package com.flutter.gradle
+import com.android.build.api.dsl.ApplicationDefaultConfig
+import com.android.build.api.dsl.ApplicationExtension
+import com.flutter.gradle.BaseApplicationNameHandler.Companion.GRADLE_BASE_APPLICATION_NAME_PROPERTY
+import org.gradle.api.Project
+import org.gradle.api.plugins.ExtensionContainer
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.mockito.Mockito
+import kotlin.test.Test
+
+class BaseApplicationNameHandlerTest {
+    @Test
+    fun setBaseName_respectsFlutterToolProperty() {
```

Comment:
Kotlin supports this:

```suggestion
    fun `setBaseName respects Flutter tool property`() {
```

it's a matter of taste but I think it's cool.

(not sure if there are some android-specific problems with it though)

Reply:
that's awesome, changed!

Still need to get these tests hooked up to run in CI, that is the last part of the PR. Working on it now.

FYI that they are runnable (with coverage even) in Android studio now through the ui